From: David Daney <ddaney@caviumnetworks.com>
Date: Wed, 17 Dec 2008 21:28:39 +0000 (-0800)
Subject: MIPS: MIPS64R2: Fix buggy __arch_swab64
X-Git-Url: http://www.linux-mips.org/git?p=linux.git;a=commitdiff_plain;h=83c620e2243110b5f6e952186d7fc4294ce3fb16

MIPS: MIPS64R2: Fix buggy __arch_swab64

The way the code is written it was assuming dshd has the function of a
hypothetical dshw instruction ...

Signed-off-by: David Daney <ddaney@caviumnetworks.com>
Signed-off-by: Ralf Baechle <ralf@linux-mips.org>
(cherry picked from commit 21faa50ed9c335bdf6ad843dcc874eb00c6ea8bb)
---

diff --git a/include/asm-mips/byteorder.h b/include/asm-mips/byteorder.h
index fe7dc2d..ddd7543 100644
--- a/include/asm-mips/byteorder.h
+++ b/include/asm-mips/byteorder.h
@@ -43,9 +43,8 @@ static __inline__ __attribute_const__ __u32 ___arch__swab32(__u32 x)
 static __inline__ __attribute_const__ __u64 ___arch__swab64(__u64 x)
 {
 	__asm__(
-	"	dsbh	%0, %1			\n"
-	"	dshd	%0, %0			\n"
-	"	drotr	%0, %0, 32		\n"
+	"	dsbh	%0, %1\n"
+	"	dshd	%0, %0"
 	: "=r" (x)
 	: "r" (x));
 
